tests/test_classifiers.py

```diff
@@ -0,0 +1,323 @@
+
+    expected_pred = cudf.Series([False])
+
+    assert result_pred.equals(expected_pred)
+
+
+@pytest.mark.gpu
+def test_multilingual_domain_classifier(gpu_client):
+    from nemo_curator.classifiers import MultilingualDomainClassifier
+
+    text = [
+        # Chinese
+        "量子计算将彻底改变密码学领域。",
+        # Spanish
+        "Invertir en fondos indexados es una estrategia popular para el crecimiento financiero a largo plazo.",
+        # English
+        "Recent advancements in gene therapy offer new hope for treating genetic disorders.",
+        # Hindi
+        "ऑनलाइन शिक्षण प्लेटफार्मों ने छात्रों के शैक्षिक संसाधनों तक पहुंचने के तरीके को बदल दिया है।",
+        # Bengali
+        "অফ-সিজনে ইউরোপ ভ্রমণ করা আরও বাজেট-বান্ধব বিকল্প হতে পারে।",
+    ]
+    df = cudf.DataFrame({"text": text})
+    input_dataset = DocumentDataset(dask_cudf.from_cudf(df, npartitions=1))
+
+    classifier = MultilingualDomainClassifier()
+    result_dataset = classifier(dataset=input_dataset)
+    result_pred = result_dataset.df.compute()["domain_pred"]
+
+    expected_pred = cudf.Series(
+        [
+            "Science",
+            "Finance",
+            "Health",
+            "Jobs_and_Education",
+            "Travel_and_Transportation",
+        ]
+    )
+
+    assert result_pred.equals(expected_pred)
+
+
+@pytest.mark.gpu
+def test_content_type_classifier(gpu_client):
+    from nemo_curator.classifiers import ContentTypeClassifier
+
+    text = ["Hi, great video! I am now a subscriber."]
+    df = cudf.DataFrame({"text": text})
+    input_dataset = DocumentDataset(dask_cudf.from_cudf(df, npartitions=1))
+
+    classifier = ContentTypeClassifier()
+    result_dataset = classifier(dataset=input_dataset)
+    result_pred = result_dataset.df.compute()["content_pred"]
+
+    expected_pred = cudf.Series(["Online Comments"])
+
+    assert result_pred.equals(expected_pred)
+
+
+@pytest.mark.gpu
+def test_prompt_task_complexity_classifier(gpu_client):
+    from nemo_curator.classifiers import PromptTaskComplexityClassifier
+
+    text = ["Prompt: Write a Python script that uses a for loop."]
+    df = cudf.DataFrame({"text": text})
+    input_dataset = DocumentDataset(dask_cudf.from_cudf(df, npartitions=1))
+
+    classifier = PromptTaskComplexityClassifier()
+    result_dataset = classifier(dataset=input_dataset)
+    result_pred = result_dataset.df.compute().sort_index(axis=1)
+
+    expected_pred = cudf.DataFrame(
+        {
+            "constraint_ct": [0.5586],
+            "contextual_knowledge": [0.0559],
+            "creativity_scope": [0.0825],
+            "domain_knowledge": [0.9803],
+            "no_label_reason": [0.0],
+            "number_of_few_shots": [0],
+            "prompt_complexity_score": [0.2783],
+            "reasoning": [0.0632],
+            "task_type_1": ["Code Generation"],
+            "task_type_2": ["Text Generation"],
+            "task_type_prob": [0.767],
+            "text": text,
+        }
+    )
```

Comment:
This passes on my remote machine (both my conda setup and my Docker setup) but is failing on our gpuCI. I think perhaps it is related to floating point errors on different machines? I will probably just add some rounding estimations so that it passes both locally for me and here on GitHub.

Reply:
cc @VibhuJawa